### ISSUE TITLE
Still confused although I know the number of classes is different

### ISSUE THREAD

**Issue by lsy641** [SenticNet/conv-emotion#14]
I know the number of classes is differrent, but you all **train on 7 classes**, right?
They use 4 classes on test  phase , and you use all classes, but you should get same results on same labels. In other words , your paper should get same value of UWA on label "happy","sad","neutral","anger". But in fact they are much more lower than CMN paper 
 
It's a little wierd that the two results are too different, although more labels used in you experiment.

Proposing this question is to consult the DialogueRNN's performance on imbalance dataset, I test DialogueRNN on EmotionLines, it even worse than CNN+biLSTM

**Comment by lsy641** [comment]
hope not close this issue so fast, I'am afraid that I still have many questions to inquire you. Thanks

**Comment by soujanyaporia** [comment]
When you change the number of classes the performance will change because you are changing the weights in the softmax layer. I hope it is clear now?

You will need to use classweight when you apply DialogueRNN on an imbalanced dataset.

**Comment by lsy641** [comment]
..Sorry, I am not  much clear.
So you mean CMN paper use 4 classes softmax in **train** phase ?
if it is true, they should delete the data of labels not in those 4 classes??

**Comment by lsy641** [comment]
ok  I know what you mean now, just neglect above

**Comment by lsy641** [comment]
So , how do you think this augment : DialogueRNN does not perform better than CNN-BiLSTM on imbalance dataset?
I think in you paper'result, it is can be seen.
![image](https://user-images.githubusercontent.com/26696711/67137452-20a95680-f268-11e9-9a06-a366513a7bcb.png)
LSTM method is more smooth on all labels.
It is also witnessed in my experiment of testing your model  on EmotionLines. It's not better than CNN+BiLSTM

**Comment by devamanyu** [comment]
Hi @lsy641, 

Let me try to clarify your doubts. In general, LSTM+att networks do provide good baselines in many applications, including ERC. However, to improve further, we look for more expressive networks with logical inductive bias (network design). The associated trade-off is that these networks become more prone to overfitting and sensitive towards label imbalances. Thus, sophisticated approaches (such as class weights, etc.) have to be adopted. In the right setting, which is normally found via an in-depth hyperparameter search, these networks have the potential to succeed over the LSTM baselines. Our results in the above-cited image demonstrate that on IEMOCAP and AVEC.

**Comment by lsy641** [comment]
> Hi @lsy641,
> 
> Let me try to clarify your doubts. In general, LSTM+att networks do provide good baselines in many applications, including ERC. However, to improve further, we look for more expressive networks with logical inductive bias (network design). The associated trade-off is that these networks become more prone to overfitting and sensitive towards label imbalances. Thus, sophisticated approaches (such as class weights, etc.) have to be adopted. In the right setting, which is normally found via an in-depth hyperparameter search, these networks have the potential to succeed over the LSTM baselines. Our results in the above-cited image demonstrate that on IEMOCAP and AVEC.

Thanks a lot @devamanyu

**Comment by lsy641** [comment]
Sorry, I come back and carry a new question.
What is the exact meaning of accuracy reported in your table below?
![image](https://user-images.githubusercontent.com/26696711/67140735-1ac76b80-f290-11e9-8690-e128679cc9e6.png)
In my comprehension of accuracy , it is formulated as (TP+TN)/(TP+TN+FP+FN)。Why the accuracy of all labels so low in your table? I think they all should be more than 80, even 90. Because TN will contribute most of the accuracy even if  TP is little.
@devamanyu @soujanyaporia   Thank you for your patience !!

**Comment by soujanyaporia** [comment]
Could you please run the models once and see the confusion matrix? That should clear your doubts and a faster way too.

**Comment by lsy641** [comment]
I guess it is unweighted average recall （UAR）or unweighted average presicion（UAP）.  And I cant't assure which one further  .
